Functions/Mock.ps1

```diff
@@ -1423,16 +1484,15 @@ function Get-DynamicParametersForCmdlet {
         return
     }
 
+    if ($null -eq $InputArgs -or $InputArgs.Count -eq 0) {
+        return
+    }
+
     if ('5.0.10586.122' -lt $PSVersionTable.PSVersion) {
         # Older version of PS required Reflection to do this.  It has run into problems on occasion with certain cmdlets,
         # such as ActiveDirectory and AzureRM, so we'll take advantage of the newer PSv5 engine features if at all possible.
 
-        if ($null -eq $Parameters) {
-            $paramsArg = @()
-        }
-        else {
-            $paramsArg = @($Parameters)
-        }
+        $paramsArg = $InputArgs | ForEach-Object { $_.ToString() }
 
         $command = $ExecutionContext.InvokeCommand.GetCommand($CmdletName, [System.Management.Automation.CommandTypes]::Cmdlet, $paramsArg)
```

Comment:
By passing all the input received from the `$args` variable instead of just the `$PSBoundParameters` it fixes the issue of restricting the parameter set based on the dynamic parameters themselves, since the `$args` variable will contain all the dynamic parameters specified that we want the original function to resolve